### PR TITLE
Code Freeze 2

### PR DESCRIPTION
**Pull request by Green-Ranger11**
# CODE FREEZE 2

### What we managed to accomplish this week

- Update main repo README to reflect our project. @prakharrathi25 did some great work on this and it looks awesome and is also informative. Check it out [here!](https://github.com/MLH-Fellowship/pod-3.1.3-team-3/tree/staging)
- Change PR template location so that it automatically shows when creating a new PR.
- Create a Callback function on sign in to get access token to authenticate octokit so we can user information from the github api. This access token is received upon authentication with github so we store it in NextJS session so we can use to make requests to github api.
- Refactored parts of the code to use next/link and other areas such as our api routes where unnecessary variables are declared. Thanks to @anandrajaram21 who made our code look so much more leaner and cleaner after his refactors.
- Move from octokit rest to core. @robzwolf thanks for the heads on this! We like it so much more!
- Add redirect on sign in and log out. Sign in will redirect to dashboard and log out will redirect to home. Challenging part is that the callback function for redirect applies to both sign in and sign out. Of course you can declare the callback manually when its being called but thats more tedious.
- Fixed Vercel deploy issues (Yay no more errors) Thanks to @anandrajaram21 for getting rid of those annoying deploy errors and its nice to get those sweet green ticks again after making a PR.
- Added requests to github api to return issues and pull requests for the signed in user. Also fixed existing requests to return accurate and relevant data. We have almost every piece of data we need! All that's left now is to display it to the user. Great job to @prakharrathi25 for getting started on all of our github requests.
- Also managed to get user private repository information.
- Added a sidebar with user profile information with navigation links.

#### Last Week
Last Week
#### This Week
d

### CODE REVIEW

**Comment by anandrajaram21** [comment]
@Green-Ranger11 Can you change the screenshot for this week? My latest PR into staging looks very different

**Comment by Green-Ranger11** [comment]
> @Green-Ranger11 Can you change the screenshot for this week? My latest PR into staging looks very different

I've changed it, what do you think?

**Comment by anandrajaram21** [comment]
Hmm, the screenshot still doesn't show all the issues and PRs. Can you use the picture I put in #63?

**Comment by Green-Ranger11** [comment]
> Hmm, the screenshot still doesn't show all the issues and PRs. Can you use the picture I put in #63?

Oh I see I see. I don't think I have any issues and prs which is why I'm empty. Anyways have used your PR

File: README.md [hunk 1]
```diff
@@ -1,11 +1,83 @@
-# Explorer Template
 
-This is a template to help you get started with your projects. Find out more information inside of the Fello Handbook.
+![pod-3.1.3-team-3](https://socialify.git.ci/MLH-Fellowship/pod-3.1.3-team-3/image?descriptionEditable=Maintain%20peak%20productivity%20by%20knowing%20exactly%20what%20you%20need%20to%20work%20on%20when%20you%20need%20to%20work%20on%20it.&font=Rokkitt&issues=1&language=1&owner=1&pattern=Brick%20Wall&pulls=1&stargazers=1&theme=Light)
+
+[![forthebadge](https://forthebadge.com/images/badges/built-by-developers.svg)](https://forthebadge.com)
+[![forthebadge](https://forthebadge.com/images/badges/made-with-javascript.svg)](https://forthebadge.com)
+# GitDash
```
Comment (robzwolf) [inline README.md:6]: I love badges!

File: README.md [hunk 2]
```diff
@@ -1,11 +1,83 @@
...
+```bash
+cd gitdash
+npm install
+```
+
+Then, run the development server:
+
+```bash
+npm run dev
+```
```
Comment (robzwolf) [inline README.md:33]: This is some great markdown. Love the inclusion of `bash` to get some syntax highlighting in there!

File: README.md [hunk 3]
```diff
@@ -1,11 +1,83 @@
...
+
+### Github Authentication
```
Comment (robzwolf) [inline README.md:35]: `Github` ➡️ `GitHub` ;)

File: gitdash/components/card.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  Tag,
+  useColorModeValue,
+  Button,
+  Link,
+} from "@chakra-ui/react";
+
+function extractRepoName(issueUrl: string) {
+  var url = new URL(issueUrl);
+  var pathArray = url.pathname.split("/");
+  return pathArray[1] + "/" + pathArray[2];
```
Comment (robzwolf) [inline gitdash/components/card.tsx:16]: Using `+` works, although I tend to prefer using [template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) in JavaScript these days.

File: gitdash/pages/api/all_repos.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+): Promise<any> {
+  const session = await getSession({ req });
+
+  const octokit = new Octokit({
+    auth: session?.accessToken,
+  });
+
+  // Get user data
+  const userData = await octokit.request("GET /user");
+  const username = userData.data.login
```
Comment (robzwolf) [inline gitdash/pages/api/all_repos.ts:30]: You should be consistent about whether or not your lines end in a semicolon `;`

File: gitdash/pages/api/github.ts [hunk 6]
```diff
@@ -1,58 +1,88 @@
...
-  const reposStarred = await octokit.request(`/users/${username}/starred`)
-  const starredCount = reposStarred.data.length
+  // Number of stars
+  // Note this really needs some way of optimizing instead of getting all the repos, parsing
+  // and adding. GraphQL!!!!!!!! We need this optimisation in a lot of places 
+  const starsCount = repos.data
+    .filter((repo: { fork: any }) => !repo.fork)
+    .reduce((acc: any, item: { stargazers_count: any }) => {
+      return acc + item.stargazers_count;
+    }, 0);
```
Comment (robzwolf) [inline gitdash/pages/api/github.ts:49]: ES6 at its finest! 🤩

File: gitdash/pages/api/github.ts [hunk 7]
```diff
@@ -1,58 +1,88 @@
...
-    return acc + item.open_issues
-  }, 0)
+  // Making the request to get the response :/
+  const reposStarred = await octokit.request("GET /user/starred?per_page=1");
 
-  // List of repos 
-  const repo_names = repos.data.map((repo: { name: any }) => repo.name)
+  // Parsing the starred repos using a library
+  const linkHeader = reposStarred.headers.link;
+  const parsed = linkHeader ? parse(linkHeader) : parse('');
```
Comment (robzwolf) [inline gitdash/pages/api/github.ts:56]: Ternary operators rock 👌

File: gitdash/pages/api/github.ts [hunk 8]
```diff
@@ -1,58 +1,88 @@
...
+  const starredCount = parsed?.last.page;
```
Comment (robzwolf) [inline gitdash/pages/api/github.ts:57]: Seeing you use the `?` here is almost enough to make me switch from JavaScript to TypeScript for all my personal projects

File: gitdash/pages/api/pulls.ts [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+      );
+
+      // Iterate through the pullData
+      for (let pull of pullData.data) {
+        allPullData.push(pull);
+      }
+    }
+  }
+  console.log("Actual data");
+  console.log(allPullData);
```
Comment (robzwolf) [inline gitdash/pages/api/pulls.ts:47]: You can combine these into command, i.e.

```javascript
console.log('Actual data:', allPullData);
```

File: gitdash/pages/api/single_repo.ts [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+          commits: any; 
+          forks: any; 
+        }): any;
+        new (): any;
+      };
+    };
+  }
+): Promise<any> {
+  const session = await getSession({ req });
+  const {repo} = req.query
```
Comment (robzwolf) [inline gitdash/pages/api/single_repo.ts:29]: Think about setting up ESLint or something like that to encourage you to have a consistent code style.

Personally, I'm a fan of the [Airbnb style guide](https://github.com/airbnb/javascript).

**Review by robzwolf — APPROVED** [review]
You've all done some absolutely fantastic work this week!

There's some seriously impressive work in here. I'm impressed by how you've used the Octokit library and your code is generally written to a very high standard.

I left a few comments about minor things – mostly code style. If you have the time, look into setting up a linter (such as ESLint).

Also, I'm a huge advocate of WebStorm (it's [free for students](https://www.jetbrains.com/community/education/#students)) for writing JS/TS. It picks up on loads of things you haven't thought about (e.g. unused imports) and helps you write code _faster_.

**Comment by robzwolf** [comment]
git merge

File: gitdash/pages/api/all_repos.ts (hunk 5 above)
Comment (prakharrathi25) [reply gitdash/pages/api/all_repos.ts:30]: Python programmer messing up consistent semi-colons, my bad. I will fix this.

File: gitdash/pages/api/github.ts (hunk 6 above)
Comment (prakharrathi25) [reply gitdash/pages/api/github.ts:49]: Thanks!

**Pull request merged**